File: Backend/src/models/chat_request.py
from pydantic import BaseModel
from typing import List, Optional
from models.chat_message import ChatMessage

class ChatRequest(BaseModel):
    messages: List[ChatMessage]
    conversation_id: Optional[int] = None  # None means create new conversation
    model: str = "gpt-4o-mini"  # Default model
    temperature: float = 0.7

File: Frontend/src/Models/ChatRequest.ts
export class ChatRequest {
    public messages: Array<{ role: string; content: string }>;
    public conversation_id?: number | null;
    public model?: string;
    public temperature?: number;

    public constructor(
        messages: Array<{ role: string; content: string }>,
        conversation_id: number | null = null,
        model: string = "gpt-4o-mini",
        temperature: number = 0.7
    ) {
        this.messages = messages;
        this.conversation_id = conversation_id;
        this.model = model;
        this.temperature = temperature;
    }
}
